cmd/datool/datool.go

```diff
@@ -0,0 +1,194 @@
+	// TODO ECDSA private key to sign message with
+	ConfConfig conf.ConfConfig `koanf:"conf"`
+}
+
+func parseClientStoreConfig(args []string) (*ClientStoreConfig, error) {
+	f := flag.NewFlagSet("client", flag.ContinueOnError)
+	f.String("url", "", "URL of DAS server to connect to.")
+	f.String("message", "", "Message to send.")
+	f.Duration("das-retention-period", 24*time.Hour, "The period which DASes are requested to retain the stored batches.")
+	conf.ConfConfigAddOptions("conf", f)
+
+	k, err := util.BeginCommonParse(f, args)
+	if err != nil {
+		return nil, err
+	}
+
+	var config ClientStoreConfig
+	if err := util.EndCommonParse(k, &config); err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+func startClientStore(args []string) error {
+	config, err := parseClientStoreConfig(args)
+	if err != nil {
+		return err
+	}
+
+	client, err := dasrpc.NewDASRPCClient(config.URL)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	cert, err := client.Store(ctx, []byte(config.Message), uint64(time.Now().Add(config.DASRetentionPeriod).Unix()))
+	if err != nil {
+		return err
+	}
+
+	serializedCert := das.Serialize(*cert)
+	encodedCert := make([]byte, base64.StdEncoding.EncodedLen(len(serializedCert)))
+	base64.StdEncoding.Encode(encodedCert, serializedCert)
+
+	fmt.Printf("Base64 Encoded Cert: %s\n", string(encodedCert))
+
+	return nil
+}
+
+// datool client retrieve
+
+type ClientRetrieveConfig struct {
+	URL        string          `koanf:"url"`
+	Cert       string          `koanf:"cert"`
+	ConfConfig conf.ConfConfig `koanf:"conf"`
+}
+
+func parseClientRetrieveConfig(args []string) (*ClientRetrieveConfig, error) {
+	f := flag.NewFlagSet("client", flag.ContinueOnError)
+	f.String("url", "", "URL of DAS server to connect to.")
+	f.String("cert", "", "Base64 encodeded DAS certificate of message to retrieve.")
+	conf.ConfConfigAddOptions("conf", f)
+
+	k, err := util.BeginCommonParse(f, args)
+	if err != nil {
+		return nil, err
+	}
+
+	var config ClientRetrieveConfig
+	if err := util.EndCommonParse(k, &config); err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+func startClientRetrieve(args []string) error {
+	config, err := parseClientRetrieveConfig(args)
+	if err != nil {
+		return err
+	}
+
+	client, err := dasrpc.NewDASRPCClient(config.URL)
+	if err != nil {
+		return err
+	}
+
+	decodedCert := make([]byte, base64.StdEncoding.DecodedLen(len(config.Cert)))
+	_, err = base64.StdEncoding.Decode(decodedCert, []byte(config.Cert))
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	message, err := client.Retrieve(ctx, decodedCert)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Message: %s\n", message)
+	return nil
+}
+
+// das keygen
+
+type KeyGenConfig struct {
+	Dir        string
+	ConfConfig conf.ConfConfig `koanf:"conf"`
+}
+
+func parseKeyGenConfig(args []string) (*KeyGenConfig, error) {
+	f := flag.NewFlagSet("client", flag.ContinueOnError)
```

Comment:
How about `datool keygen` instead of just `client`?

Reply:
I made this change in https://github.com/OffchainLabs/nitro/pull/589